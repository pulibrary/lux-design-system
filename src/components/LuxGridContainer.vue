<template>
  <component :is="type" :class="['lux-flex-container', horizontal]">
    <!-- @slot The grid items that live in your grid container. -->
    <slot />
  </component>
</template>

<script>
/**
 * Used to build the container of GridItems.
 */
export default {
  name: "LuxGridContainer",
  status: "ready",
  release: "1.0.0",
  type: "Element",
  props: {
    /**
     * The html element name used for the grid container.
     */
    type: {
      type: String,
      default: "div",
    },
    /**
     * Determines how the flexbox container is horizontally aligned. This value defaults to
     * having space between grid items. Options include `start`, `center`, and `end`.
     */
    horizontal: {
      type: String,
      default: "",
    },
  },
}
</script>

<style lang="scss" scoped>
.lux-flex-container {
  padding: 0 1rem 1rem 0;
  display: flex;
  flex-wrap: wrap;
  justify-content: space-between;
  width: 100%;

  &.start {
    justify-content: flex-start;
  }

  &.center {
    justify-content: center;
  }

  &.end {
    justify-content: flex-end;
  }
}
</style>

<docs>
  ```jsx
    <div>
      <lux-grid-container>Grid container can be used to contain grid items together.</lux-grid-container>
    </div>
  ```
</docs>
