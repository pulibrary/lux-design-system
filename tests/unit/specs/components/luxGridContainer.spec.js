import { mount } from "@vue/test-utils"
import LuxGridContainer from "@/components/LuxGridContainer.vue"

describe("GridItem.vue", () => {
  let wrapper

  beforeEach(() => {
    wrapper = mount(LuxGridContainer, {
      props: {
        horizontal: "center",
      },
    })
  })

  it("can take the horizontal alignment from the props", () => {
    expect(wrapper.find("div").classes()).toContain("center")
  })

  it("has the expected html structure", () => {
    expect(wrapper.element).toMatchSnapshot()
  })
})
