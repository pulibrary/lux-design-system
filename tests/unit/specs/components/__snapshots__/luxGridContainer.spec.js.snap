// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GridItem.vue has the expected html structure 1`] = `
<div
  class="lux-flex-container center"
>
  <!-- @slot The grid items that live in your grid container. -->
  
  
</div>
`;
